DataFormats/FWLite/src/DataGetterHelper.cc

```diff
@@ -131,20 +133,22 @@ namespace fwlite {
         //obj.destruct();
         ////END OF WORK AROUND
 
-        TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
+        if (tcUse_) {
+            TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
 
-        if (nullptr == tcache) {
-            iData.branch_->GetEntry(eventEntry);
-        } else {
-            if (!tcTrained_) {
-                tcache->SetLearnEntries(100);
-                tcache->SetEntryRange(0, tree_->GetEntries());
-                tcTrained_ = true;
+            if (nullptr != tcache) {
+                if (!tcTrained_) {
+                    tcache->SetLearnEntries(100);
+                    tcache->SetEntryRange(0, tree_->GetEntries());
+                    tcTrained_ = true;
+                }
+                tree_->LoadTree(eventEntry);
             }
-            tree_->LoadTree(eventEntry);
-            iData.branch_->GetEntry(eventEntry);
-       }
-       iData.lastProduct_=eventEntry;
+        }
+        branchAccFoo_(iData.branch_);
```

Comment:
Is the need to tune the TTreeCache specific to Fireworks?  If this is something other applications could benefit from, could we encapsulate the operations in some kind of cache policy object that gets invoked at the appropriate places, rather than having Fireworks manage its own TTreeCache and getting a branch access callback?